### PR TITLE
[Spec] Support sampling in the DSPARK graph-folded draft proposal

### PR DESCRIPTION
**Pull request by hnyls2002**
Extend the graph-folded DSPARK draft proposal from greedy-only to mixed greedy + sampling batches: the per-step fused kernel argmaxes greedy rows and Gumbel-samples the rest with in-graph philox noise, exporting the corrected block logits for the sampling accept path (which stays eager). The required static noise/logits buffers are gated by SGLANG_DSPARK_FOLDED_SAMPLING (OFF / AUTO / FORCE, default AUTO): AUTO allocates them only when they fit in the free GPU memory left after target init, otherwise falls back to today's greedy-only folding.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #30776635445](https://github.com/sgl-project/sglang/actions/runs/30776635445)
Latest PR Test (Extra): :x: [Run #30776635323](https://github.com/sgl-project/sglang/actions/runs/30776635323)

The folded proposal head (greedy_step_sampler / DsparkDraftSampler / maybe_build_draft_sampler) moves byte-identically into its own module `dspark_draft_sampler.py`, so `dspark_draft.py` keeps only the proposal orchestration and its diff stays small.

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test test_basic_sanity_dspark.py test_dspark_kernel_parity.py test_dspark_stacked_ctx_kv_parity.py test_dspark_draft_path_default.py test_dspark_scheduler.py test_dspark_block_accept_estimator.py test_dspark_confidence_metrics.py test_dspark_dp_tier.py test_dspark_info_dumper.py test_dspark_sps_profiler.py test_dspark_sps_table.py test_dspark_sts.py test_ragged_verify.py test_ragged_verify_backend_capability.py test_kimi_linear_dcp_dspark4.py
